Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM osrf/ros:humble-desktop-full
+
+# Install dependencies (taken from devcontainer dockerfile)
+RUN DEBIAN_FRONTEND=noninteractive apt-get update -q && \
+    apt-get update -q && \
+    apt-get install -yq --no-install-recommends \
+    lcov \
+    curl \
+    python3-pip \
+    python-is-python3 \
+    python3-argcomplete \
+    python3-colcon-common-extensions \
+    python3-colcon-mixin \
+    python3-rosdep \
+    libpython3-dev \
+    python3-vcstool \
+    ros-humble-tl-expected && \
+    rm -rf /var/lib/apt/lists/*
+
+WORKDIR /repo
+
+COPY . .
+
+RUN sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules
+RUN git submodule update --init --recursive
```

Comment:
I think you will need to carry these lines over from the .devcontainer dockerfile to install the boston dynamics related requirements 

```suggestion
RUN git submodule update --init --recursive

# Install bosdyn_msgs package
RUN curl -sL https://github.com/bdaiinstitute/bosdyn_msgs/releases/download/4.0.2/ros-humble-bosdyn_msgs_4.0.2-jammy_amd64.run --output /tmp/ros-humble-bosdyn_msgs_4.0.2-jammy_amd64.run --silent \
  && chmod +x /tmp/ros-humble-bosdyn_msgs_4.0.2-jammy_amd64.run \
  && ((yes || true) | /tmp/ros-humble-bosdyn_msgs_4.0.2-jammy_amd64.run) \
  && rm /tmp/ros-humble-bosdyn_msgs_4.0.2-jammy_amd64.run

# Install spot_cpp_sdk package
RUN curl -sL https://github.com/bdaiinstitute/spot-cpp-sdk/releases/download/v4.0.2/spot-cpp-sdk_4.0.2_amd64.deb --output /tmp/spot-cpp-sdk_4.0.2_amd64.deb --silent \
  && dpkg -i /tmp/spot-cpp-sdk_4.0.2_amd64.deb \
  && rm /tmp/spot-cpp-sdk_4.0.2_amd64.deb

# Install bosdyn_msgs missing dependencies
RUN python -m pip install --no-cache-dir --upgrade pip==22.3.1 \
    && pip install --root-user-action=ignore --no-cache-dir --default-timeout=900 \
    numpy==1.24.1 \
    pytest-cov==4.1.0 \
    pytest-xdist==3.5.0 \
    bosdyn-api==4.0.2 \
    bosdyn-core==4.0.2 \
    bosdyn-client==4.0.2 \
    bosdyn-mission==4.0.2 \
    bosdyn-choreography-client==4.0.2 \
    pip cache purge

# Install spot_wrapper requirements
RUN --mount=type=bind,source=spot_wrapper,target=/tmp/context/spot_wrapper cd /tmp/context && \
    pip install --root-user-action=ignore --no-cache-dir --default-timeout=900 -r spot_wrapper/requirements.txt && \
    pip cache purge
```

Reply:
Interesting. I realized that I copied the repository code into the root of the ros workspace rather than the source directory, so I need to update the dockerfile regardless. I'll see what's going on.

I was able to run the dockerfile as-is and connect to my robot with no other steps. I'm as confused as you are as to how...

Reply:
As for the bosdyn python packages, those get taken care of by the `install_spot_ros2.sh` script when it installs the requirements for `spot_wrapper`

Reply:
No worries.

This dockerfile is just meant to provide a turn-key docker image for the driver that can be used as a base for projects using it in workflows like my robotics lab's where we use docker containers to manage the switching between different ROS installs (rather than switching operating systems whenever we need to jump between major ROS versions)